Dockerfile.dev

```diff
@@ -0,0 +1,9 @@
+FROM node:latest
+RUN mkdir /usr/src/app
+WORKDIR /usr/src/app
+ENV PATH /usr/src/app/node_modules:$PATH
+RUN npm install
```

Comment:
We'll need to copy over the package*.json files to the image first.

Reply:
I think I might handle this in the `./start init` script instead and save `node_modules` to a volume.  I think that prevents us needing to rebuild the image when node_modules change.  Does that make sense or am I talking crazy?

Reply:
Also, I don't understand why this currently works :joy: It definitely seems like it should fail.

Reply:
You're already creating a volume for the npm modules in your docker compose. This volume _should_ be cached and not reinstalled every build. Unless you delete the volume. You probably don't want to pull out this step 😄 If you're pushing this container to heroku it won't be able to get the modules

It probably works on your machine because you have the packages on your local machine that are getting pulled into the image due to the volume `".:/usr/src/app"`. `rm -rf ./node_modules` 😉

Reply:
My understanding is that we won't need _both_ the `RUN npm install` and volume in `docker-compose`.  The former bakes the modules into the image and the latter allows them to be installed and cached at run-time.  Since this stuff is for dev only, the latter seems to make the most sense, though we'll need to trigger an `npm install` at some point during the start-up phase (maybe in `docker-entrypoint.sh`?)

Production assets are built by using the separate `Docker.build` file and is really only necessary in order to bundle it with `nginx`.  I'm not convinced that's even the best way to go about organizing the deploy :stuck_out_tongue: It's still sort of left over from when we were deploying to `now` thru `drone` so it might be worth tearing out and doing something more sane.  That seems like it belongs in a separate PR, though.

Reply:
I think these, in the end, accomplish the same thing for most intents and purposes. However there is a benefit of running it in the Dockerfile in that it will be cached as an intermediate container (idk the actual technical term lol) in addition to being available as a volume at run time. So if for whatever reason you did a `docker-compose down -v`, you'd still have that cached layer when you rebuild the image. 

I personally usually have these build steps in the Dockerfile.

Reply:
Ah yeah, I s'pose you're right 🤔 I'm gonna merge for now and if we find out there's a better way to do it in the future, we'll PR it :+1: